src/battle_dynamax.c

```diff
@@ -884,12 +884,15 @@ void BS_TrySetStatus1(void)
             }
             break;
         case STATUS1_SLEEP:
-            if (CanBeSlept(gBattlerTarget, GetBattlerAbility(gBattlerTarget)))
+            if (CanBeSlept(gBattlerTarget, GetBattlerAbility(gBattlerTarget), TRUE))
             {
                 if (B_SLEEP_TURNS >= GEN_5)
                     gBattleMons[gBattlerTarget].status1 |=  STATUS1_SLEEP_TURN((Random() % 3) + 2);
                 else
                     gBattleMons[gBattlerTarget].status1 |=  STATUS1_SLEEP_TURN((Random() % 4) + 3);
+                
+                // Try to activate Sleep Clause when a mon is put to Sleep
```

Comment:
I'm not going to fight but is there a chance we can remove all those `Try Activate / Deactivate` comments? They don't tell me more then what the code already does in my view.

Reply:
i can see why the comments wouldnt tell _you_ specifically any more info, but i would say the reason for that is partially due to the fact that you are much more familiar with the battle code than the average expansion user :D i also dont really have a strong opinion on this matter and if we really want to ill delete the comments, but i also would scratch my head a bit at the notion of explicitly removing information and documentation. it doesnt seem like it's doing any harm in being there and there's a non-zero chance _someone_ would benefit from it

Reply:
I think I agree with Alex on this one, "Try to active Sleep Clause" is `TryActivateSleepClause` and "when a mon is put to Sleep" is implied by `CanBeSlept` followed by setting `STATUS1_SLEEP_TURN`.

The latter is certainly a little subtle, in that I could believe somebody reads that line of code and doesn't immediately figure out that it's putting the 'mon to sleep; but I think we assume that the reader understands what's going on everywhere else.

I don't know how to distill my intuition about which comments pull their weight down into an ironclad guideline, but I think I'd set the line somewhere around "comments aimed at a person who can read code and understands the Pokémon mechanic being implemented but is reading quickly and out of context so might miss subtle details about the implementation (especially those details that are only relevant elsewhere)".

That said I also don't care to fight on it :)